### PR TITLE
Observable module

### PR DESCRIPTION
**Pull request by Olllom**
Separate the computation of observables from reporting.
Breaking change: Deleted `GenericStepReporter` and subclasses in lieu of `ObservableReporter`

New usage example:

```
from lettuce import TaylorGreenVortex3D, Enstrophy, D3Q27, Lattice
lattice = Lattice(D3Q27, device="cpu")
flow = TaylorGreenVortex(50, 300, 0.1, lattice)
enstrophy = Enstrophy(lattice, flow)
reporter = ObservableReporter(enstrophy, interval=10)
```

The advantage is that observables can also be computed outside a reporter. This is important, for example when we want to backpropagate through an observable in a loss function.

@MartinKliemank @McBs @dominikwilde  This will break the old API. Please take note.

### CODE REVIEW

**Comment by McBs** [comment]
Good change that avoids the need for double implementation which as it has been done so far.
